### PR TITLE
Remove support for Ruby <2.7

### PR DESCRIPTION
**Pull request by szh**
### Desired Outcome

dev/Dockerfile.dev in conjur-api-ruby uses the ruby:2.5 base image. This image includes a number of known vulnerabilities. Update this to a later version (preferably one that matches our upgraded Ruby version everywhere else).

### Implemented Changes

- Remove support for Ruby versions below 2.7, which are [end of life](https://endoflife.date/ruby).

### Connected Issue/Story

CyberArk internal issue link: [ONYX-16435](https://ca-il-jira.il.cyber-ark.com:8443/browse/ONYX-16435)

#### Changelog

- [x] The CHANGELOG has been updated, or
- [ ] This PR does not include user-facing changes and doesn't require a
  CHANGELOG update

#### Test coverage

- [ ] This PR includes new unit and integration tests to go with the code
  changes, or
- [x] The changes in this PR do not require tests

#### Documentation

- [ ] Docs (e.g. `README`s) were updated in this PR
- [ ] A follow-up issue to update official docs has been filed here: [insert issue ID]()
- [x] This PR does not require updating any documentation

#### Behavior

- [ ] This PR changes product behavior and has been reviewed by a PO, or
- [ ] These changes are part of a larger initiative that will be reviewed later, or
- [x] No behavior was changed with this PR

#### Security

- [ ] Security architect has reviewed the changes in this PR,
- [ ] These changes are part of a larger initiative with a separate security review, or
- [ ] There are no security aspects to these changes

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -123,11 +123,8 @@ $ docker-compose down
 ### Update the version and changelog
 
 1. Create a new branch for the version bump.
-1. Based on the unreleased content, determine the new version number and update
-    the [version.rb](lib/conjur-api/version.rb) file.
 1. Commit these changes - `Bump version to x.y.z` is an acceptable commit message - and open a PR
-   for review. Your PR should include updates to `lib/conjur-api/version.rb`, and
```
Comment (john-odonnell) [inline CONTRIBUTING.md:129]: I think this entire `Releasing` section needs some more attention. The project was recently added to the new automated release flow, but this section was never updated to reflect that.

**Review by andytinkham — APPROVED** [review]
LGTM

**Review by john-odonnell — APPROVED** [review]
LGTM 👍  Left a note about updating CONTRIB to reflect the new auto-release flow, but that doesn't really affect the scope of these security changes.

File: CONTRIBUTING.md (hunk 1 above)
Comment (szh) [reply CONTRIBUTING.md:129]: Agreed. I just removed one inaccuracy but it definitely needs a larger overhaul.

**Pull request merged**